src/mpid/ch4/netmod/ucx/ucx_win.h

```diff
@@ -226,7 +226,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_NM_rma_target_local_cmpl_hook(int rank, MPIR_
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_NM_RMA_TARGET_LOCAL_CMPL_HOOK);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_NM_RMA_TARGET_LOCAL_CMPL_HOOK);
 
-    if (MPIDI_UCX_is_reachable_target(rank, win, MPIDI_WIN(win, winattr)) &&
+    if (MPIDI_UCX_is_reachable_target(rank, win, MPIDI_WIN(win, winattr), NULL) &&
         MPIDI_UCX_WIN(win).target_sync[rank].need_sync == MPIDI_UCX_WIN_SYNC_FLUSH_LOCAL) {
```

Comment:
It might be cleaner if we could separate the checking of `origin_addr` from that of target side checking. In other words, `MPIDI_UCX_is_reachable_target` checks only the target side conditions (e.g., whether target win buffer is zero-size or is from GPU mem). In this way, we don't need set the confusing `NULL` parameter in the cmpl_hook calls.

Reply:
Yeah, I suppose the API doesn't really match. Since `ch4:ucx` only supports put and get right now, this should not be too bad to leave separate.